hw1/02-palindrome.js

```diff
@@ -1,3 +1,45 @@
 const elem = document.querySelector('input');
+const errorElem = document.querySelector('.inputError');
+const resultElem = document.querySelector('.displayResult');
 
 elem.addEventListener('input', handleInput);
+
+function handleInput() {
+  const inputValue = elem.value;
+  const numValue = Number(inputValue);
+
+  // Clear the error and border when input is empty
+  if (!inputValue) {
+    resultElem.textContent = '';
+    errorElem.textContent = '';
+    elem.style.borderColor = '';
+    return;
+  }
+
+  // Check if the input is a negative number
+  if (numValue < 0) {
+    errorElem.style.color = 'red';
+    elem.style.borderColor = 'red';
+    errorElem.textContent = 'Enter a valid positive number.';
+    resultElem.textContent = '';
+    return;
+  }
+
+  // Reset error and check for palindrome
+  errorElem.textContent = '';
+  elem.style.borderColor = '';
+
+  if (isPalindrome(inputValue)) {
```

Comment:
Linter:
 32:9   error  'isPalindrome' was used before it was defined  no-use-before-define

Reply:
Thanks Sarah! I updated the code accordingly to your feedback.